### PR TITLE
feat(streaming): add `must_output_per_barrier` flag for stream simple agg

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- fix https://github.com/risingwavelabs/risingwave/issues/18055#issuecomment-2325326447 introduced in https://github.com/risingwavelabs/risingwave/pull/17945
- revert #18292 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/executor/simple_agg.rs [hunk 1]
```diff
@@ -481,4 +498,98 @@ mod tests {
...
+        // No stream chunk
+        // let msg = simple_agg.next().await.unwrap().unwrap();
+        // assert_eq!(
+        //     *msg.as_chunk().unwrap(),
+        //     StreamChunk::from_pretty(
+        //         "  I   I   I  I
+        //         U- 0   .   .  .
+        //         U+ 0   .   .  ."
+        //     )
+        // );
```
Comment (fuyufjh) [inline src/stream/src/executor/simple_agg.rs:574]:
```suggestion
```

File: src/stream/src/executor/simple_agg.rs [hunk 2]
```diff
@@ -481,4 +498,98 @@ mod tests {
...
```
Comment (fuyufjh) [inline src/stream/src/executor/simple_agg.rs:589]:
```suggestion
```

**Review by fuyufjh — APPROVED** [review]
LGTM

**Comment by st1page** [comment]
https://buildkite.com/risingwave-test/ch-benchmark-pg-cdc/builds/390

**Review by kwannoel — APPROVED** [review]

**Comment by StrikeW** [comment]
Hi, could you add some background of the cause?👀

**Comment by KeXiangWang** [comment]
Thanks for the effort! https://buildkite.com/risingwave-test/upgrade/builds/62 The upgrade test (backward compatibility test) passed.

**Pull request merged**